### PR TITLE
Java code style configuration file for Android Studio

### PR DESCRIPTION
**Pull request by ArazAbishov**
Code style which is based on default IntelliJ's settings and those from checkstyle configuration. Issue: https://github.com/artem-zinnatullin/qualitymatters/issues/201

### CODE REVIEW

File: code_quality_tools/idea-codestyle.xml [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+      <emptyLine />
+      <package name="javax" withSubpackages="true" static="false" />
+      <emptyLine />
+      <package name="" withSubpackages="true" static="false" />
+      <emptyLine />
+      <package name="" withSubpackages="true" static="true" />
+      <emptyLine />
+    </value>
+  </option>
+  <option name="RIGHT_MARGIN" value="100" />
```
Comment (artem-zinnatullin) [inline code_quality_tools/idea-codestyle.xml]: Can you please increase this to 150 or something like this?

**Comment by artem-zinnatullin** [comment]
cc @vanniktech PTAL :)

File: code_quality_tools/idea-codestyle.xml (hunk 1 above)
Comment (vanniktech) [inline code_quality_tools/idea-codestyle.xml]: Yeah 150 sounds better

**Comment by vanniktech** [comment]
I guess it's looking good. I'm using square formatter

File: code_quality_tools/idea-codestyle.xml (hunk 1 above)
Comment (ArazAbishov) [inline code_quality_tools/idea-codestyle.xml]: Updated PR.

File: code_quality_tools/idea-codestyle.xml [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+  <AndroidXmlCodeStyleSettings>
+    <option name="USE_CUSTOM_SETTINGS" value="true" />
+  </AndroidXmlCodeStyleSettings>
+  <Objective-C-extensions>
```
Comment (ArazAbishov) [inline code_quality_tools/idea-codestyle.xml]: Nice catch. AS exporter included configuration for objective c for some reason. Removed in update.

**Comment by serj-lotutovici** [comment]
@artem-zinnatullin I would advice against that... If you have plugins installed that add some code style of their own, it will always edit that file...

**Comment by ArazAbishov** [comment]
I also tend to think it will be better to keep code style settings outside of .idea folder.

**Comment by artem-zinnatullin** [comment]
@serj-lotutovici @ArazAbishov then it makes almost no sense to me because nobody will apply it, you can always undo changes in this file before commiting!

We use this at work and don't have problem you described with plugins.

**Comment by ArazAbishov** [comment]
@artem-zinnatullin I have tried to put `codeStyle.xml` under `.idea/` directory and then to open project in AS in order to check if settings are applied automatically. Instead of applying settings, AS just replaced the content of the whole folder. I have tried to name file as `codeStyleSettings.xml`, same result.

**Comment by artem-zinnatullin** [comment]
We have it in .idea/codeStyleSettings.xml and don't have such problems, can
you try to do that and reimport project?

On 17 Oct 2016 11:45 am, "Araz Abishov" <notifications@github.com> wrote:

@artem-zinnatullin <https://github.com/artem-zinnatullin> I have tried to
put codeStyle.xml under .idea/ directory and then to open project in AS in
order to check if settings are applied automatically. Instead of applying
settings, AS just replaced the content of the whole folder. I have tried to
name file as codeStyleSettings.xml, same result.

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/artem-zinnatullin/qualitymatters/pull/206#issuecomment-254148248>,
or mute the thread

**Comment by ArazAbishov** [comment]
@artem-zinnatullin I have tried to put it under .idea with corresponding .gitignore and codeStyleSettings.xml. Whenever you are trying to re-import / open the project, studio just erases the whole directory and creates new set of files.

**Comment by artem-zinnatullin** [comment]
Hm, but it works for several projects I'm part of…

**Comment by serj-lotutovici** [comment]
@ArazAbishov the contents for the file should be like this:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<project version="4">
  <component name="ProjectCodeStyleSettingsManager">
    <option name="PER_PROJECT_SETTINGS">
      <value>
        <!-- Here you put the code style xml. -->
        <!-- Just put everything embedded in the <code_scheme name="quality-matters"> block -->
      </value>
    </option>
    <option name="USE_PER_PROJECT_SETTINGS" value="true" />
  </component>
</project>
```

**Comment by ArazAbishov** [comment]
@serj-lotutovici Thanks! @artem-zinnatullin Pushed patch with settings under .idea (including .gitignore).

**Pull request merged**